.sync/github_templates/contributing/CONTRIBUTING.md

```diff
@@ -46,6 +46,49 @@ If you cannot find an existing issue that describes your bug or feature, create
 Please continue to follow your request after it is submitted to assist with any additional information that might be
 requested.
 
+### Pull Request Best Practices
+
+Pull requests for UEFI code can become large and difficult to review due to the large number of build and
+configuration files. To aid maintainers in reviewing your code, we suggest adhering to the following guidelines:
+
+1. Do keep code reviews single purpose; don't add more than one feature at a time.
+2. Do fix bugs independently of adding features.
+3. Do provide documentation and unit tests.
+4. Do introduce code in digestible amounts.
+   * If the contribution logically be broken up into separate pull requests that independently build and function
+     successfully, do use multiple pull requests.
+
+#### Code Categories
+
+To keep code digestible, you may consider breaking large pull requests into three categories of commits within the pull request.
+
+1. **Interfaces**: .h, .inf, .dec, documentation
+2. **Implementation**: .c, unit tests, unit test build file; unit tests should build and run at this point
+3. **Integration/Build**: .dec, .dsc, .fdf, integration tests; code added to platform and affects downstream consumers
+
+By breaking the pull request into these three categories, the pull request reviewers can digest each piece independently.
+
+If your commits are still very large after adhering to these categories, consider further breaking the pull request
+down by library/driver; break each component into its own commit.
+
+#### Implementation Limits
+
+Implementation is ultimately composed of functions as logical units of code.
+
+To help maintainers review the code and improve long-term maintainability, limit functions to 60 lines of code. If your
+function exceeds 60 lines of code, it likely has also exceeded a single responsibility and should be broken up.
+
+Furthermore, files contain functions. Files are easier to review and maintain if they contain functions that serves a

```

Comment:
Furthermore, files contain functions.

This is a fragment of a sentence. I think the fragment could be removed and still convey the message.

Reply:
That would work. Added it as a suggestion.